core/src/main/scala/chisel3/Bits.scala

```diff
@@ -185,25 +185,28 @@ sealed abstract class Bits(private[chisel3] val width: Width) extends Element wi
     *
     * @example
     * {{{
-    * myBits = 0x5 = 0b101
-    * myBits(1,0) => 0b01  // extracts the two least significant bits
+    *   val myBits = "0b101".U
+    *   myBits(1,0) // "0b01".U  // extracts the two least significant bits
+    *
+    *   // Note that zero-width ranges are also legal
+    *   myBits(-1, 0) // 0.U(0.W) // zero-width UInt
     * }}}
     * @param x the high bit
     * @param y the low bit
-    * @return a hardware component contain the requested bits
+    * @return a hardware component containing the requested bits
     */
   final def apply(x: Int, y: Int): UInt = macro SourceInfoTransform.xyArg
 
   /** @group SourceInfoTransformMacro */
   final def do_apply(x: Int, y: Int)(implicit sourceInfo: SourceInfo): UInt = {
-    if (x < y || y < 0) {
-      Builder.error(s"Invalid bit range ($x,$y)")
+    if (x < y - 1 || y < 0) {
+      Builder.error(s"Invalid bit range [$x, $y]")
     }
-    val w = x - y + 1
+    val resultWidth = x - y + 1
```

Comment:
I renamed this because there is variable name shadowing going on inside that `getOrElse` (hit expand to see it) and it confused me.